src/opm/parser/eclipse/EclipseState/Grid/FieldProps.cpp

```diff
@@ -1211,6 +1215,48 @@ bool FieldProps::tran_active(const std::string& keyword) const {
     return calculator != this->tran.end() && calculator->second.size() > 0;
 }
 
+void FieldProps::applyNumericalAquifers(const NumericalAquifers& numerical_aquifers) {
+    // TODO: ideally, we should also update the cell_depth, but it is not used for later
+    // in the simulator. So we do not update it here.
+    // Maybe also cell volume? Not sure whether it is used later
+    auto& porv_data = this->init_get<double>("PORV").data;
+    auto& satnum_data = this->int_data["SATNUM"].data;
+    auto& pvtnum_data = this->int_data["PVTNUM"].data;
+    numerical_aquifers.updateCellProps(*(this->grid_ptr), porv_data, satnum_data, pvtnum_data);
```

Comment:
I think this should be turned around - the `FieldProps` class should get the information from the aquifers and then modify the properties internally in the `FieldProps` - instead of sending mutable data out of the `FieldProps` container.

Reply:
I think it is a good suggestion. Thanks.